### PR TITLE
Feature/connect admin components

### PR DESCRIPTION
**Pull request by MagdalenaGie**
Little rebiuld of the app architecture and folders structure. Also finally we've begin to use api - now the app works in authenticated / not authenticated state. I use redux for that, not sure if its the best idea, but it was the fastest and most reliable solution i could have come up with for now.

### CODE REVIEW

File: front/src/components/Navigation/NavigationLinks/NavigationLinks.js [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+            {isAuthenticated ? (
+                <Fragment>
+                    <NavigationLink link="/admin/competitions">
+                        Moje zawody
+                    </NavigationLink>
+                    <NavigationLink link="/logout">Wyloguj</NavigationLink>
+                </Fragment>
+            ) : (
+                <NavigationLink link="/login">Zaloguj się</NavigationLink>
+            )}
```
Comment (frogi16) [inline front/src/components/Navigation/NavigationLinks/NavigationLinks.js:29]: Could you extract both conditional branches into separate functions? Ternary operator is great for short texts, like single function calls or assignments, but for longer, multi-line ones, becomes clumsy.

File: front/src/App.test.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+import { createStore, applyMiddleware, compose } from "redux";
+import thunk from "redux-thunk";
 import App from "./App";
 
+//this line is for chrome devtools
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+
+//in the future possibly we would have more reducers, co we would use combineReducers, but for now there is one
+const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));
+
```
Comment (frogi16) [inline front/src/App.test.js:14]: This lines are repeating over and over, probably you need them for all frontend unit tests. If so, please, refactor it to common utils file.
This way future modifications will be easier and you won't have to import all these new dependencies in every test file.

File: front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js [hunk 3]
```diff
@@ -14,21 +14,27 @@ function CompetitionDetails(props) {
         <div>
             <p className="editLinkText">
                 <span className="panel-detail-header">Szczegóły zawodów</span>
-                <span><a
-                        href={"/admin/panel/" + _id + "/details/edit"}>
-                        edytuj
-                     </a>
+                <span>
+                    <a href={"/admin/panel/" + _id + "/details/edit"}>edytuj</a>
```
Comment (frogi16) [inline front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js:18]: These are purely whitespace changes, do you have Prettier up and running? I don't see the reason why it wouldn't format these lines.

**Review by frogi16 — CHANGES_REQUESTED** [review]
Remove **all** console.logs. We could implement a system to log only in debug mode, but right now this is a security risk.

Please, investigate this warning (Chrome 90.0) and fix it if possible:
"Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://reactjs.org/link/unsafe-component-lifecycles for details."
If not, let me know.

File: front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js (hunk 3 above)
Comment (MagdalenaGie) [reply front/src/components/AdminPanel/Panel/CompetitionDetails/CompetitionDetails.js:18]: I think this component was written by @AleksandraRolka , I don't remember making any changes there - in that case I'm nit sure whether there is something wrong with mine or her configurations. Another thing is that this line should probably be changed anyway as we cannot use <a> tags and links to navigate, as it refreshes the whole page - but I havent noticed that before, as we haven't been using redux at the beginning

File: front/src/components/LoginPanel/comp/RegisterForm.js [hunk 4]
```diff
@@ -35,11 +43,12 @@ const RegisterForm = (props) => {
 
     const handleSubmit = (event) => {
         event.preventDefault();
-        console.log(login, password1, password2);
+        console.log(login, email, password1, password2);
```
[the remark this answers is not in the document]
Comment (MagdalenaGie) [reply front/src/components/LoginPanel/comp/RegisterForm.js:46]: I'm sorry, it was only for debugging purposes and I forgot to delete that lines ;)

File: front/src/components/Navigation/NavigationLinks/NavigationLinks.js [hunk 5]
```diff
@@ -7,6 +7,20 @@ import { Fragment } from "react";
...
+                <NavigationLink link="/admin/competitions">
+                    Moje zawody
+                </NavigationLink>
+                <NavigationLink link="/logout">Wyloguj</NavigationLink>
+            </Fragment>
+        );
+    } else {
+        navLinks = <NavigationLink link="/login">Zaloguj się</NavigationLink>;
+    }
+
```
Comment (frogi16) [inline front/src/components/Navigation/NavigationLinks/NavigationLinks.js:23]: I had something like this in mind:
isAuthenticated ? linksForLoggedUser() : linksForUnknownUser()

**Review by frogi16 — APPROVED** [review]
Great work with WithProvider!
Use ternary operator like I described and you are free to merge. Remember to investigate the warning I mentioned in previous comment.

File: front/src/components/LoginPanel/LoginPanel.js [hunk 6]
```diff
@@ -27,31 +28,31 @@ class LoginPanel extends Component {
...
+                            Masz już konto?
+                            <Button
+                                className="btn btn-outline-primary"
+                                onClick={this.showLogin}
+                                placeholder="Zaloguj się!"
+                            />
+                        </Fragment>
+                    ) : (
+                        <Fragment>
+                            Nie masz jeszcz konta?
```
Comment (miczoc98) [inline front/src/components/LoginPanel/LoginPanel.js:47]: typo
```suggestion
                            Nie masz jeszcze konta?
```

**Pull request merged**